src/core/ddsi/src/q_entity.c

```diff
@@ -3311,7 +3306,7 @@ static void gc_delete_writer (struct gcreq *gcreq)
   /* Do last gasp on SEDP and free writer. */
   if (!is_builtin_entityid (wr->e.guid.entityid, NN_VENDORID_ECLIPSE))
     sedp_dispose_unregister_writer (wr);
-  whc_free (wr->whc);
+  dds_whc_free (wr->whc);
```

Comment:
Like it suggested somewhere in a comment, it’s pretty much a bug that the DDSI writer takes ownership of the WHC — or at least it is very counter-intuitive. I changed it for the RHC when I cleaned that up a bit, I think it should now also be changed for the WHC.